tsl/src/remote/data_fetcher.c

```diff
@@ -68,20 +71,18 @@ data_fetcher_get_tuple(DataFetcher *df, int row)
 	Assert(df->tuples != NULL);
 	Assert(row >= 0 && row < df->num_tuples);
 
-	return df->tuples[row];
+	ExecForceStoreHeapTuple(df->tuples[row], slot, /* shouldFree = */ false);
```

Comment:
```suggestion
	/*
	 * Return the next tuple. Must force the tuple into the slot since
	 * CustomScan initializes ss_ScanTupleSlot to a VirtualTupleTableSlot
	 * while we're storing a HeapTuple.
	 */
	ExecForceStoreHeapTuple(df->tuples[row], slot, /* shouldFree = */ false);
```